src/core/client.ts

```diff
@@ -125,7 +125,9 @@ export default abstract class Client {
   setContext(context: Record<string, unknown>): Client {
     if (typeof context === 'object') {
       const store = this.__store.getStore()
-      store.context = merge(store.context, context)
+      if (store) {
```

Comment:
In `addBreadcrumb`, you set the store `GlobalStore` if it's `undefined`. Shouldn't you do the same here? And perhaps wrap into a private function to re-use (call `this.getStore()` instead of `this._store.getStore()`)?
My concern with this approach is that we may accidentally switch to a `GlobalStore` when we originally wanted the `AsyncStore`. Then again, we [always](https://github.com/honeybadger-io/honeybadger-js/blob/master/src/server/aws_lambda.ts#L34) set the store to `AsyncStore` before we to use it.

Reply:
I did this at first, but it's slightly more complicated. I went with a `getStore()` method, but it caused trouble, as some places mutated the store, while others didn't. I couldn't simply return a copy, because some locations _wanted_ to mutate the store. For example, `addBreadcrumb()` definitely wants to add to the store, hence the switch. 

I admit that it's confusing, but I decided to go with this for now, but I'll continue looking for improvements. I'll take another look now, so you can go ahead and release the main fix in the other PR.

The place you linked where we always use the async store is in AWS Lambda handlers. In regular apps, we don't do that, because an app may or may not be a web app, and errors can happen before we start handling requests (eg in our setup code).